flixel/addons/display/FlxClothSprite.hx

```diff
@@ -0,0 +1,533 @@
+package flixel.addons.display;
+import flixel.FlxCamera;
+import flixel.FlxObject;
+import flixel.FlxSprite;
+import flixel.graphics.FlxGraphic;
+import flixel.graphics.frames.FlxFrame;
+import flixel.graphics.frames.FlxFrame.FlxFrameAngle;
+import flixel.graphics.frames.FlxFrame.FlxFrameType;
+import flixel.math.FlxPoint;
+import flixel.math.FlxRect;
+import flixel.system.FlxAssets.FlxGraphicAsset;
+import flixel.util.FlxColor;
+import flixel.util.FlxDestroyUtil;
+import flixel.util.FlxSpriteUtil;
+import openfl.display.BitmapData;
+import openfl.display.Graphics;
+import openfl.geom.Point;
+import openfl.Vector;
+
+@:keep @:bitmap("assets/images/logo/default.png")
+private class GraphicDefault extends BitmapData {}
+
+/**
+ * A FlxSprite that draw it's frame in a mesh and behave like a cloth.
+ * 
+ * @author adrianulima
+ */
+class FlxClothSprite extends FlxSprite
+{
+	/**
+	 * An extra speed applied to mesh (in pixels per second). Use to simulate gravity or wind.
+	 */
+	public var gravity(default, null):FlxPoint = FlxPoint.get();
+	/**
+	 * Determines how quickly the mesh points come to rest.
+	 */
+	public var friction(default, null):FlxPoint = FlxPoint.get(0.99, 0.99);
+	/**
+	 * Change the size of the mesh related to the original frame. Need to call setMesh() to update.
+	 */
+	public var meshScale(default, null):FlxPoint = FlxPoint.get(1, 1);
+	/**
+	 * Bit field of flags (use with FlxObject.UP, DOWN, LEFT, RIGHT, NONE, ANY, etc) indicating pinned side. Use bitwise operators to check the values stored here.
+	 */
+	public var pinSide:Int;
+	/**
+	 * How many iterations will do on constraints for each update. Need to call setMesh() to update.
+	 * Bigger number make constraint more strong and mesh more rigid.
+	 */
+	public var iterations:Int = 3;
+	/**
+	 * Adds an extra constraint crossing the squares to make the mesh more rigid. Need to call setMesh() to update.
+	 */
+	public var crossingConstraints:Bool = false;
+	/**
+	 * Number of columns of the mesh. To set it you must use setMesh().
+	 */
+	public var columns(default, null):Int;
+	/**
+	 * Number of rows of the mesh. To set it you must use setMesh().
+	 */
+	public var rows(default, null):Int;
+	/**
+	 * The width of mesh squares. To set it you must use setMesh().
+	 */
+	public var widthInTiles(default, null):Float;
+	/**
+	 * The height of mesh squares. To set it you must use setMesh().
+	 */
+	public var heightInTiles(default, null):Float;
+	
+	public var points(default, null):Array<ClothPoint> = [];
+	public var constraints(default, null):Array<ClothConstraint> = [];
+	
+	/**
+	 * Mesh arrays. Vertices, indices and uvtData to drawTriangles().
+	 */
+	#if flash
+	private var _vertices(default, null):Vector<Float>;
```

Comment:
You could use a typedef to avoid the redundancy here ([see here](https://github.com/HaxeFlixel/flixel/blob/dev/flixel/graphics/tile/FlxDrawTrianglesItem.hx#L14)).